### ISSUE TITLE
Green artifacts on videos (OpenGL + Wayland Backend only)

### ISSUE THREAD

**Issue by mrrhq** [mpv-player/mpv#2659]
Heyas, I'm one of those few people that have been running Wayland on Debian Sid and have been loving it for the past few days. But an issue has struck me when using these vo options:

```
vo=opengl:backend=wayland
```

The problem is, there are ugly green artifacts that show up while the video is playing. It may be codec specific (MPEG/MP4) but I'm not sure. It's those same green artifacts that VLC users may be familiar with. This does **not** happen with the other vo drivers I've tried, like SDL or DRM. I have not however tested another backend with OpenGL yet. I have also not tried OpenGL-HQ, since my computer is a little slow to handle those settings, but it can (I use a Core i7 laptop with internal Intel GMA graphics) I know for a fact that this does not happen with OpenGL on X11, since I've done that before.

If you guys need any system info, version info, uname output, logs or screenshots of videos, I'm happy to provide. But just try it yourself. I think it may be because I'm also using XWayland, but I don't think that mpv would start itself as an X client since I specified the Wayland backend, right?

**Comment by wm4** [comment]
Any new observations? I can't really think of anything why this would specifically happen with Wayland (and not e.g. X11, which would use the same underlying GL driver).

**Comment by haasn** [comment]
Do other OpenGL-on-Wayland applications work without any problems?

**Comment by mrrhq** [comment]
I just want to write in and say that I've also been seeing these kind of thing under X11 as well. I've narrowed it down to me using:

```
vo=xv
hwdec = auto
```

In my config file, it seems to use vaapi, so I tried other hwdec options like vaapi-copy, which doesn't work. So I just turn it off, and use xv with it, and it seems to play the video fine under X11.

I am using an X201 Tablet, and have the i965 video driver (Intel) installed for the Intel Core i7 L 620 processor on Linux.

So now, I'll try different vo and hwdec options and see what works under Wayland. I'll try video out for both SDL and OpenGL....
